be/src/storage/lake/spark_load.cpp

```diff
@@ -0,0 +1,162 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+#include "storage/lake/spark_load.h"
+
+#include "runtime/runtime_state.h"
+#include "storage/chunk_helper.h"
+#include "storage/lake/tablet_writer.h"
+#include "storage/push_handler.h"
+#include "storage/storage_engine.h"
+#include "storage/tablet_manager.h"
+#include "storage/lake/filenames.h"
+
+namespace starrocks::lake {
+
+using PushBrokerReader = starrocks::vectorized::PushBrokerReader;
+
+Status SparkLoadHandler::process_streaming_ingestion(const Tablet& tablet, const TPushReq& request, PushType push_type,
+                                                     std::vector<TTabletInfo>* tablet_info_vec) {
+    LOG(INFO) << "begin to realtime vectorized push. tablet=" << tablet.id() << ", txn_id: " << request.transaction_id;
+    DCHECK(request.__isset.use_vectorized && request.use_vectorized);
+
+    _request = request;
+    Status st = _do_streaming_ingestion(tablet, request, push_type);
+    if (st.ok()) {
+        if (tablet_info_vec != nullptr) {
+            _get_tablet_infos(tablet, tablet_info_vec);
+        }
+        LOG(INFO) << "process realtime vectorized push successfully. "
+                  << "tablet=" << tablet.id() << ", partition_id=" << request.partition_id
+                  << ", txn_id: " << request.transaction_id;
+    }
+
+    return st;
+}
+
+Status SparkLoadHandler::_do_streaming_ingestion(Tablet tablet, const TPushReq& request, PushType push_type) {
+    Status res = Status::OK();
+    PUniqueId load_id;
+    load_id.set_hi(0);
+    load_id.set_lo(0);
+
+    // create txn log
+    auto txn_log = std::make_shared<lake::TxnLog>();
+    txn_log->set_tablet_id(tablet.id());
+    txn_log->set_txn_id(request.transaction_id);
+
+    Status st = Status::OK();
+    if (push_type == PUSH_FOR_DELETE) {
+        return Status::NotSupported("does not support push delete for lake table");
+    }
+
+    if (push_type == PUSH_NORMAL_V2) {
+        st = _load_convert(tablet, request);
+    }
+
+    return Status::OK();
```

Comment:
```suggestion
```
if (push_type == PUSH_NORMAL_V2) {
    return _load_convert(tablet, request);
} else {
    return Status::NotSupported(fmt::format("unsupported push type for lake table: {}", push_type));
}
```
```